its/src/test/java/com/sonar/it/scanner/msbuild/utils/TestUtils.java

```diff
@@ -307,6 +307,15 @@ public static Path getMsBuildPath(Orchestrator orch) {
     return msBuildPath;
   }
 
+  /*
+    Use this instead, because missing component throws HTTP error without listing the actual components:
+    assertThat(TestUtils.listComponents(ORCHESTRATOR, context.projectKey))
+      .extracting(Components.Component::getKey)
+      .containsExactlyInAnyOrder(
+        ...
+      )
+   */
+  @Deprecated // ToDo: Remove in SCAN4NET-201
```

Comment:
There is a use case where you query `component:someFile.cs`. I don't know if this works with `listComponents`. See also https://github.com/SonarSource/sonar-scanner-msbuild/pull/2435#discussion_r2020532621

Reply:
It's fine. all remaining usages are easily removable.

I'll probably do it later in this PR, after it's rebased on master.

Reply:
The `SolutionKindTest` PR was merged, and this is now based on it. So I've pushed one new commit to remove this helper